benchmarks/low_level.cpp

```diff
@@ -0,0 +1,455 @@
+
+/*
+    File contains benchmarks for
+    -   transactions size [1, 4] -> [1, 2]
+    -   full tx serialization
+    -   full tx deserialization
+    -   full tx round trip
+    -   compact tx serialization
+    -   compact tx deserialization
+    -   compact tx round trip
+    -   tx creation
+    -   tx signing
+    -   tx validation
+
+
+    Note: Contains call to BENCHMARK_MAIN
+*/
+
+#include "uhs/transaction/messages.hpp"
+#include "uhs/transaction/transaction.hpp"
+#include "uhs/transaction/validation.hpp"
+#include "uhs/transaction/wallet.hpp"
+
+#include "util/serialization/istream_serializer.hpp"
+#include "util/serialization/ostream_serializer.hpp"
+
+#include <benchmark/benchmark.h>
+#include <filesystem>
+#include <gtest/gtest.h>
+
+class low_level : public ::benchmark::Fixture {
+  protected:
+    void SetUp(const ::benchmark::State&) override {
+        auto mint_tx1 = wallet1.mint_new_coins(2, 100);
+        wallet1.confirm_transaction(mint_tx1);
+        auto mint_tx2 = wallet2.mint_new_coins(1, 100);
+        wallet2.confirm_transaction(mint_tx2);
+
+        m_of.open(m_BENCHMARK_File);
+    }
+
+    void TearDown(const ::benchmark::State&) override {
+        m_if.close();
+        m_of.close();
+        std::filesystem::remove_all(m_BENCHMARK_File);
+    }
+
+    cbdc::transaction::wallet wallet1;
+    cbdc::transaction::wallet wallet2;
+
+    cbdc::transaction::full_tx m_valid_tx{};
+    cbdc::transaction::full_tx m_valid_tx_multi_inp{};
+
+    // std::unique_ptr<secp256k1_context, decltype(&secp256k1_context_destroy)>
+    //     m_secp{secp256k1_context_create(SECP256K1_CONTEXT_SIGN
+    //                                     | SECP256K1_CONTEXT_VERIFY),
+    //            &secp256k1_context_destroy};
+    // cbdc::privkey_t m_priv0{cbdc::hash_from_hex(
+    //     "0000000000000001000000000000000000000000000000000000000000000000")};
+    // cbdc::pubkey_t m_pub0{cbdc::pubkey_from_privkey(m_priv0, m_secp.get())};
+    // cbdc::privkey_t m_priv1{cbdc::hash_from_hex(
+    //     "1000000000000001000000000000000000000000000000000000000000000000")};
+    // cbdc::pubkey_t m_pub1{cbdc::pubkey_from_privkey(m_priv1, m_secp.get())};
+    // std::unordered_set<cbdc::pubkey_t, cbdc::hashing::null> m_pubkeys{m_pub0,
+    //                                                                   m_pub1};
+
+    // for serialization
+    std::ifstream m_if;
+    std::ofstream m_of;
+
+    cbdc::istream_serializer m_is{m_if};
+    cbdc::ostream_serializer m_os{m_of};
+
+    static constexpr auto m_BENCHMARK_File
+        = "serial_BENCHMARK_File.dat";
+};
+
+// create_tx 1-1
+BENCHMARK_F(low_level, create_1to1_tx)(benchmark::State& state) {
+    // if(state.thread_index() == 0) {
+    //     GTEST_LOG_(INFO) << "Starting transaction benchmarks ... ";
+    // }
+    for(auto _ : state) {
+        m_valid_tx
+            = wallet1.send_to(100, wallet2.generate_key(), true).value();
```

Comment:
`wallet::send_to()` just generates a transaction (might be a good candidate for a better name in the future), the result needs to be sent somewhere for it to matter. As a result, there's no need for the confirm-and-send-back pattern.

The one thing we would want to double-check is to ensure that the tx-creation/assignment isn't elided due to it not being used. But, that could just be an assert checking that the transaction is structured as you expect (1:1).